tsnet/example/tsnet-funnel/tsnet-funnel.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+// The tsnet-funnel server demonstrates how to use tsnet with funnel.
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+
+	"tailscale.com/ipn"
+	"tailscale.com/tsnet"
+)
+
+var (
+	addr = flag.String("addr", ":443", "address to listen on")
+)
+
+func enableFunnel(ctx context.Context, s *tsnet.Server) error {
+	st, err := s.Up(ctx)
+	if err != nil {
+		return err
+	}
+	if len(st.CertDomains) == 0 {
+		return errors.New("tsnet: you must enable HTTPS in the admin panel to proceed")
+	}
+	domain := st.CertDomains[0]
+
+	hp := ipn.HostPort(net.JoinHostPort(domain, "443"))
+
+	srvConfig := &ipn.ServeConfig{
+		AllowFunnel: map[ipn.HostPort]bool{
+			hp: true,
+		},
+	}
+	lc, err := s.LocalClient()
+	if err != nil {
+		return err
+	}
+	return lc.SetServeConfig(ctx, srvConfig)
```

Comment:
What do you think about rolling this up into `tsnet.Server` functions like `FunnelOn(hp ipn.HostPort) error` and `FunnelOff(hp ipnHostPort) error`.

Reply:
yeah, something like that would be nice! Can do it in a follow up change so that its isolated and we can figure out the appropriate API.